docs/security-features/privilege-restriction/cgroups.rst

```diff
@@ -0,0 +1,104 @@
+- ``blkio``: Manages block I/O, controlling read/write access to block devices like hard drives and SSDs.
+- ``pids``: Limits the number of processes that can be created within a cgroup, providing a crucial defence against "fork bomb" attacks.
+- ``network_cls``/``network_prio``: Used to tag network packets from processes within a cgroup, allowing for traffic shaping and prioritization.
+
+More information about controllers can be found `here <https://www.kernel.org/doc/html/latest/admin-guide/cgroup-v2.html>`_.
+
+
+The Role of Cgroups in System Stability and Security
+----------------------------------------------------
+
+While often seen as a resource management tool, cgroups are a critical component of system
+security and stability.
+
+Preventing Resource Exhaustion
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The primary security benefit of cgroups is their ability to prevent any single process or service
+from monopolizing system resources. A buggy application with a memory leak or a compromised service
+attempting a denial-of-service attack cannot consume all available memory or CPU time if a cgroup
+constrains it. This containment ensures that the rogue process will be terminated or throttled 
+before it can crash the entire system, preserving the availability of other essential services.
+
+Foundations of Containerization
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Cgroups are the bedrock of container technologies like Docker, LXD, and Snap. They provide the
+resource isolation that makes containers viable. By placing each container into its own cgroup,
+the system can guarantee that one container cannot access more than its allocated share of CPU
+or memory, effectively preventing it from interfering with other containers or the host system
+itself.
+
+Service Management with ``systemd``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+On modern Ubuntu systems, ``systemd`` makes extensive use of cgroups to manage all system services.
+Every service started by ``systemd`` is automatically placed into its own dedicated cgroup. This
+provides several benefits: 
+
+- When a service is stopped, ``systemd`` can reliably terminate the main process of its children
+  by killing all tasks within the service's cgroup.
+- ``systemd`` can accurately track the resource consumption of each service. Tools like 
+  ``systemd-cgtop`` provides a real-time view of CPU, memory, and I/O usage, broken down by service.
+- Administrators can easily apply resource limits directly to a service using:
+
+  .. code-block:: sh
+ 
+    sudo systemctl set-property <target-service> <property>
+
+  For example, an admin can limit the memory available to a database server
+  to prevent it from impacting the rest of the system under heavy load.
+
+
+
+Inspecting Cgroups on Ubuntu
+----------------------------
+
+Any interested user can directly explore their cgroup hierarchy, which is mounted as a virtual
+filesystem at :file:`/sys/fs/cgroup`. To see the cgroup a specific service belongs to, a
```

Comment:
I think we should reference `/proc/<pid>/cgroup` here, as well as `systemctl status <pid>`. Bonus points if we can explain why `systemctl` only shows one name but the procfs interface shows a dozen.